pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -172,6 +227,11 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 		return fmt.Errorf("failed to ensure finalizer to %s from cluster %s: %w", isName.String(), srcClusterName, err)
 	}
 
+	key := simpleImageStream{NamespacedName: types.NamespacedName{Namespace: req.Namespace, Name: imageStreamName}}
```

Comment:
This should be at the very beginning of the reconciliation

Reply:
Updated.